src/sliding_sync.rs

```diff
@@ -2143,9 +2189,8 @@ fn handle_sync_service_state_subscriber(mut subscriber: Subscriber<sync_service:
             log!("Received a sync service state update: {state:?}");
             if state == sync_service::State::Error {
                 log!("Restarting sync service due to error.");
-                if let Some(ss) = SYNC_SERVICE.get() {
-                    ss.start().await;
-                }
+                let sync_service = get_sync_service().expect("BUG: sync service is None");
+                sync_service.start().await;
```

Comment:
if we panic here (via the `.expect()` call), then the entire main background task will crash and nothing else will work.

I think that this is a good place to show a user-visible error, like so:
```suggestion
                if let Some(ss) = get_sync_service() {
                    ss.start().await;
                } else {
                    enqueue_popup_notification(PopupItem {
                        message: "Unable to restart the Matrix sync service.\n\nPlease quit and restart Robrix.".into(),
                        auto_dismissal_duration: None,
                        kind: PopupKind::Error,
                    });
                }
```